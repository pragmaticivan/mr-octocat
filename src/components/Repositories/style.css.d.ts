export const repo_card_list: string;
export const repoCardList: string;
export const search_wrapper: string;
export const searchWrapper: string;
