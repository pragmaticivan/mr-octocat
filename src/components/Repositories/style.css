
.repo_card_list {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
}

.repo_card_list > div {
  width: calc(100% * (1/2) - 34px - 6px);
}
