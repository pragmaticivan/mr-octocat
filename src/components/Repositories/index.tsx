import Button from "../Button";
import InputSearch from "../InputSearch";
import React from "react";
import RepoCardFull from "../RepoCardFull";
import style from "./style.css";

interface Props {
  repositories: [];
}

export default class Repositories extends React.Component<Props> {
  render() {
    const { repositories } = this.props;
    return (
      <div>
        <div className={style.searchWrapper}>
          <div className={style.inputWrapper}>
            <InputSearch />
          </div>
          <div className={style.typeWrapper}>
            <Button>Type: All v</Button>
          </div>
          <div className={style.languageWrapper}>
            <Button>Language: All v</Button>
          </div>
        </div>
        <div className={style.repoCardList}>
          {repositories.map((repo: any) => {
            return <RepoCardFull repository={repo} key={repo.id} />;
          })}
        </div>
      </div>
    );
  }
}
