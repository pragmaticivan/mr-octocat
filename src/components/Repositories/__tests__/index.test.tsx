import Repositories from "../";
import React from "react";
import { shallow } from "enzyme";

describe("Repositories", () => {
  it("renders the component", () => {
    const repository = {
      created_at: "2016-08-02T17:35:14Z",
      description:
        "Language Savant. If your repository's language is being reported incorrectly, send us a pull request!",
      fork: true,
      forks: 53,
      forks_count: 53,
      full_name: "octocat/linguist",
      html_url: "https://github.com/octocat/linguist",
      id: 64778136,
      language: "Ruby",
      name: "linguist",
      stargazers_count: 17,
      updated_at: "2019-05-26T09:08:16Z",
    };
    const wrapper = shallow(<Repositories repositories={[repository]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
