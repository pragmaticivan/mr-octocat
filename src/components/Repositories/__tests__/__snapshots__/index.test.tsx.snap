// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repositories renders the component 1`] = `ShallowWrapper {}`;
